src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -190,4 +201,43 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    public void setOnButtonPressedListener() {
+        commandBox.getCommandTextField().setOnKeyPressed(event -> {
+            switch (event.getCode()) {
+            case UP:
+                int newIndex = (selectedIndex - 1);
+                if (newIndex < 0) {
+                    newIndex = 0;
+                }
+                selectedIndex = newIndex;
+
+                System.out.println("PRESSED UP current index" + newIndex);
```

Comment:
Note System.out.println, may want to use Logger instead? This is more a nitpick though.